lib/cadence/saga/concern.rb

```diff
@@ -14,9 +14,21 @@ def run_saga(&block)
         logger.error("Saga execution aborted: #{error.inspect}")
         logger.debug(error.backtrace.join("\n"))
 
-        saga.compensate
+        if compensable?(error, configuration)
+          logger.error('Saga compensating')
+          saga.compensate
+          Result.new(false, error)
+        else
+          logger.error('Saga not compensating')
+          raise error
+        end
+      end
 
-        Result.new(false, error)
+      def compensable?(error, compensable_errors: nil, non_compensable_errors: nil)
```

Comment:
nit: why not default these to `[]` and avoid `.nil?`s and `&.`s below?

Reply:
By defaulting to nil I was thinking about the case where an empty array could be passed in which would signal the caller never wants to compensate

Reply:
Actually, that would be better achieved via a flag passed into the `run_saga` configuration